#ifndef C_DIALOG_ANIMATIONS_H
#define C_DIALOG_ANIMATIONS_H

//****************************************************************************************************
//Класс диалога настройки анимации
//****************************************************************************************************

//****************************************************************************************************
//подключаемые библиотеки
//****************************************************************************************************
#include <QDialog>
#include "cimagestorage.h"
#include "ctilessequence.h"

//****************************************************************************************************
//пространства имён
//****************************************************************************************************

namespace Ui
{
 class CDialog_Animations;
}

//****************************************************************************************************
//макроопределения
//****************************************************************************************************

//****************************************************************************************************
//константы
//****************************************************************************************************

//****************************************************************************************************
//предварительные объявления
//****************************************************************************************************

//****************************************************************************************************
//Класс диалога настройки анимации
//****************************************************************************************************
class CDialog_Animations:public QDialog
{
 Q_OBJECT
 public:
  //-перечисления---------------------------------------------------------------------------------------
  //-структуры------------------------------------------------------------------------------------------
  //-константы------------------------------------------------------------------------------------------
 private:
  //-константы------------------------------------------------------------------------------------------
  static const size_t CYCLYC_ANIMATION_INDEX=0;//индекс циклической анимации
  static const size_t SET_STEP_ANIMATION_INDEX=1;//индекс анимации с заданием шага
  static const size_t ONE_SHOT_ANIMATION_INDEX=2;//индекс однократной анимации
  //-переменные-----------------------------------------------------------------------------------------
  Ui::CDialog_Animations *ui;

  int32_t SelectedTileIndexX;//позиция выбранного тайла по X
  int32_t SelectedTileIndexY;//позиция выбранного тайла по Y

  int32_t SelectedTileSequenceIndexX;//позиция выбранного тайла в последовательности по X
  int32_t SelectedTileSequenceIndexY;//позиция выбранного тайла в последовательности по Y

  CTilesSequence cTilesSequence;//последовательность анимации
 public:
  //-конструктор----------------------------------------------------------------------------------------
  CDialog_Animations(QWidget *parent,const CTilesSequence &cTilesSequence_set);
  //-деструктор-----------------------------------------------------------------------------------------
  ~CDialog_Animations();
 public:
  //-открытые функции-----------------------------------------------------------------------------------
  CTilesSequence& GetTilesSequence(void);//получить последовательность анимации
 private:
  //-закрытые функции-----------------------------------------------------------------------------------
  void UpdateTilesImage(void);//обновить картинку списка тайлов
  bool SelectedTile(QPoint &mpos);//выбрать тайл в наборе
  bool SelectedTileSequence(QPoint &mpos);//выбрать тайл в последовательности
  void mousePressEvent(QMouseEvent *qMouseEvent_Ptr);//обработчик нажатия на кнопку мышки
  void mouseReleaseEvent(QMouseEvent *qMouseEvent_Ptr);//обработчик отпускания кнопки мышки
 private slots:
  void on_cPushButton_Add_clicked(void);//слот нажатия на кнопку добавление тайла в последовательность
  void on_cPushButton_Clear_clicked(void);//слот нажатия на кнопку удаления всех тайлов кроме первого из последовательности
  void on_cPushButton_Delete_clicked();//слот нажатия на кнопку удаления выбранного тайла из последовательности
  void on_cPushButton_Cancel_clicked();//слот нажатия на кнопку отмена
  void on_cPushButton_Apply_clicked();//слот нажатия на кнопку применить
};

#endif
